packages/components/src/circular-option-picker/option-picker-option.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { useContext, useEffect } from '@wordpress/element';
+import { Icon, check } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { CircularOptionPickerContext } from './option-picker-context';
+import Button from '../button';
+import { CompositeItem } from '../composite';
+import Tooltip from '../tooltip';
+import type { OptionProps } from './types';
+
+const hasSelectedOption = new Map();
+
+function renderOptionAsButton( props: {
+	className?: string;
+	isPressed?: boolean;
+} ) {
+	return <Button { ...props }></Button>;
+}
+
+function renderOptionAsOption( props: {
+	id: string;
+	className?: string;
+	isSelected?: boolean;
+	compositeState: any;
+} ) {
+	const { className, isSelected, compositeState, ...additionalProps } = props;
+
+	return (
+		<CompositeItem
+			{ ...additionalProps }
+			{ ...compositeState }
+			as={ Button }
+			className={ classnames( className, {
+				'is-pressed': isSelected,
+			} ) }
+			role="option"
+			aria-selected={ !! isSelected }
+		/>
+	);
+}
+
+export function Option( {
+	className,
+	isSelected,
+	selectedIconProps,
+	tooltipText,
+	...additionalProps
+}: OptionProps ) {
+	const compositeState = useContext( CircularOptionPickerContext );
+	const { baseId, currentId, setCurrentId } = compositeState as any;
+	const isComposite = !! baseId;
```

Comment:
This check feels brittle — why don't we add a `isComposite` prop to the `CircularOptionPickerContext` object?